### PR TITLE
[vs17.2] Customize opt-in for CG step

### PR DESCRIPTION
**Pull request by JanKrivanek**
### Context
Explict call to `ComponentGovernanceComponentDetection@0` task (via `/eng/common/templates/steps/component-governance.yml` template) based on the branch name (so that this can be invoked for 'vs*' branches as well)

### Note
This will be breaking until we consume this arcade update: https://github.com/dotnet/arcade/commit/b1a9b866bd8e28f55a68b4048306bd1ccb3acb45, it's now flowing to MSBuild main in this PR: https://github.com/dotnet/msbuild/pull/8658

Testing the changes on: https://github.com/dotnet/msbuild/tree/vs-test-only
ADO run: (vs* branch, the CG step injected) https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7672204&view=logs&j=bb592630-4b9d-53ad-3960-d954a70a95cf&t=424211a8-7b86-5e93-500e-ae39b50be1a7

### Blocked by 
https://github.com/dotnet/msbuild/pull/8697
https://github.com/dotnet/msbuild/pull/8302

### CODE REVIEW

**Review by wtgodbe — APPROVED** [review]

**Comment by JanKrivanek** [comment]
> LGTM! I'm assuming it would be syntactically incorrect to split the 'if' across multiple lines with indentations, correct? Just wondering because it's a bit hard to parse the dis/conjunctions 🙂

Yeah - syntactic meaning of indention and whitespaces is comming into way. There might be a way - I just didn't find any example in whole arcade and wasn't able to quickly invent something myself.

For reference:
 * run with formatted condition (failed due to script syntax): https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7698171&view=results
 * run with single line (hardly readable) condition (run, failed due to unrelated issues): https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7698211&view=logs&j=bb592630-4b9d-53ad-3960-d954a70a95cf&t=94418e61-6648-5751-f7d4-a14f4e5e2bb7

**Comment by JanKrivanek** [comment]
This needs update of arcade to `6.0.0-beta.23221.7`
